### ISSUE TITLE
language-server can't find rubocop

### ISSUE THREAD

**Issue by breathe** [rubyide/vscode-ruby#552]
Issue Type: <b>Bug</b>

From Ruby Language Server output:

```
Lint: executing bundle exec rubocop -s /Users/ncohen/software/Web/app/models/location.rb -f json...
Lint: unable to execute bundle exec rubocop -s /Users/ncohen/software/Web/app/models/location.rb -f json as the command could not be found
```

I get a same error with or without setting the rubocop useBundler flag:

```
    "ruby.lint": {
      "rubocop": {
        "useBundler": true
      }
    },
```

Works from command line (with or without bundle exec)
```
ncohen@208 ~/s/i/Web> cat /Users/ncohen/software/Web/app/models/location.rb | rubocop -s /Users/ncohen/software/Web/app/models/location.rb -f json
{"metadata":{"rubocop_version":"0.71.0","ruby_engine":"ruby","ruby_version":"2.4.6","ruby_patchlevel":"354","ruby_platform":"x86_64-darwin18"},"files":[{"path":"app/models/location.rb","offenses":[]}],"summary":{"offense_count":0,"target_file_count":1,"inspected_file_count":1}}
ncohen@208 ~/s/i/Web> cat /Users/ncohen/software/Web/app/models/location.rb | bundle exec rubocop -s /Users/ncohen/software/Web/app/models/location.rb -f json
{"metadata":{"rubocop_version":"0.71.0","ruby_engine":"ruby","ruby_version":"2.4.6","ruby_patchlevel":"354","ruby_platform":"x86_64-darwin18"},"files":[{"path":"app/models/location.rb","offenses":[]}],"summary":{"offense_count":0,"target_file_count":1,"inspected_file_count":1}}
```

```
The file must be piped into rubocop or other utilities. This methodology is the best representation of how the language server runs these commands. If the command succeeds here but not in the language server, additional steps will be necessary to troubleshoot. Please open an issue.
```

Hmm -- can't figure this out and opening an issue as the troubleshooting docs suggest.  Much appreciate!

Extension version: 0.25.3
VS Code version: Code 1.39.2 (6ab598523be7a800d7f3eb4d92d7ab9a66069390, 2019-10-15T15:33:00.827Z)
OS version: Darwin x64 18.7.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 x 2800)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (0.84GB free)|
|Process Argv|itsacheckmate.code-workspace|
|Screen Reader|no|
|VM|0%|
</details>

**Comment by andrewmcodes** [comment]
What’s the output of `rubocop -v`?

Can you paste your config relating to this extension?

**Comment by breathe** [comment]
```
➤ rubocop -v
0.71.0
```

In workspace file are all the settings related to vscode-ruby -- and I've tried all the different variations of the "ruby.useBundler" and "ruby.lint" options ...
```
  "settings": {
    "ruby.useLanguageServer": true,

    "[javascript]": {
      "editor.defaultFormatter": "esbenp.prettier-vscode",
      "editor.formatOnSave": true
    },
    "[typescript]": {
      "editor.defaultFormatter": "esbenp.prettier-vscode"
    },
    "[json]": {
      "editor.defaultFormatter": "esbenp.prettier-vscode"
    },
    "[ruby]": {
      "editor.formatOnSave": true,
      "editor.defaultFormatter": "jnbt.vscode-rufo",
      "editor.formatOnSaveTimeout": 3000
    },
    "ruby.lint": {
      "rubocop": true
    }
  }
```

**Comment by breathe** [comment]
I'm using fish shell as well if there's any chance that might impact behavior here ...?

**Comment by andrewmcodes** [comment]
@breathe First, I would try `gem install rubocop` just to get on the latest version, then modify your config slightly to resemble this: 

```
  "settings": {
    "[ruby]": {
      "editor.formatOnSave": true,
    },
    "ruby.useLanguageServer": true,
    "ruby.lintDebounceTime": 1500,
    "ruby.lint": {
      "rubocop": {
         "useBundler": true
     }
    "ruby.format": "rubocop"
  }
```

**Comment by ajimix** [comment]
I'm having the same problem. It seems to be related with fish as I also use fish: https://github.com/rubyide/vscode-ruby/issues/523

Seems like we have to wait for that issue to be closed first before using this extension

**Comment by wingrunr21** [comment]
Or just set up a minimal setup that is POSIX compliant just to run RuboCop. Fish support is in progress.

**Comment by e1senh0rn** [comment]
Warning: the following is a hack and it will be overwritten on vscode-ruby update:

You could edit `~/.vscode/extensions/rebornix.ruby-0.25.3/client/out/util/shims/env.fish.sh` to this:
```
#!/usr/local/bin/fish

for name in (set -nx)
  if string match --quiet '*PATH' $name
    echo $name=(string join : -- $$name)
  else
    echo $name="$$name"
  end
end
```

**Comment by ajimix** [comment]
> Warning: the following is a hack and it will be overwritten on vscode-ruby update:
> 
> You could edit `~/.vscode/extensions/rebornix.ruby-0.25.3/client/out/util/shims/env.fish.sh` to this:
> 
> ```
> #!/usr/local/bin/fish
> 
> for name in (set -nx)
>   if string match --quiet '*PATH' $name
>     echo $name=(string join : -- $$name)
>   else
>     echo $name="$$name"
>   end
> end
> ```

this worked for me but in the following file instead (I'm on Mac using VSCodium, maybe that is why the file is different) `~/.vscode-oss/extensions/rebornix.ruby-0.25.3/dist/extension/shims/env.fish.sh`

**Comment by e1senh0rn** [comment]
@ajimix my bad, you are correct. I reused path to file from earlier vscode-ruby version.

**Comment by wingrunr21** [comment]
Fish shell support for the environment detection landed in `v0.26.0`. Please give it a try!
